### PR TITLE
Add tests for mimetypes returned by content-app

### PR DESCRIPTION
**Pull request by lubosmj**
Required PR: https://github.com/pulp/pulpcore/pull/3093
[noissue]

### CODE REVIEW

File: pulp_file/tests/functional/api/from_pulpcore/test_content_types.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    file_repo_api_client,
+    file_repo_with_auto_publish,
+    file_content_unit_with_name,
+    gen_object_with_cleanup,
+):
+    """Test if content-app correctly returns mime-types based on filenames."""
+    files = {
+        "tar.gz": file_content_unit_with_name(f"{str(uuid.uuid4())}.tar.gz"),
+        "xml.gz": file_content_unit_with_name(f"{str(uuid.uuid4())}.xml.gz"),
+        "json.zstd": file_content_unit_with_name(f"{str(uuid.uuid4())}.json.zstd"),
```
Comment (dralley) [inline pulp_file/tests/functional/api/from_pulpcore/test_content_types.py:25]: Let's include 

* standard `.json`
* `.xml.bz2`
* `.xml.zstd`
* `.xml.xz`

File: pulp_file/tests/functional/api/from_pulpcore/test_content_types.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import aiohttp
```
Comment (bmbouter) [inline pulp_file/tests/functional/api/from_pulpcore/test_content_types.py:1]: Please move this to be `pulp_file/tests/functional/api/test_mime_types.py`. We're moving all code out of the `from_pulpcore` directory, it was the landing area for code copied from pulpcore only.

File: pulp_file/tests/functional/conftest.py [hunk 3]
```diff
@@ -66,6 +66,17 @@ def file_random_content_unit(
     return gen_object_with_cleanup(file_content_api_client, **artifact_attrs)
 
 
+@pytest.fixture
```
Comment (bmbouter) [inline pulp_file/tests/functional/conftest.py:69]:
```suggestion
def file_content_unit_with_name_factory(
```

Anytime the factory pattern is used we've started adding `_factory` to the end to clearly indicate it's usage as a factory.

File: pulp_file/tests/functional/conftest.py [hunk 4]
```diff
@@ -66,6 +66,17 @@ def file_random_content_unit(
...
+def file_content_unit_with_name_factory(
+    file_content_api_client, tmp_path, random_artifact, gen_object_with_cleanup
+):
+    def _file_content_unit_with_name(name):
```
Comment (bmbouter) [inline pulp_file/tests/functional/conftest.py:73]: This should match also please.

**Review by bmbouter — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Review by dralley — APPROVED** [review]

**Pull request merged**